src/test/java/uk/gov/pay/directdebit/pact/PublicApiContractTest.java

```diff
@@ -136,4 +141,26 @@ public void aDirectDebitEventExists(Map<String, String> params) {
                 .withExternalId(params.getOrDefault("external_id", RandomIdGenerator.newId()))
                 .insert(app.getTestContext().getJdbi());
     }
+
+    @State("a gateway account with external id and a confirmed mandate with a payment on it exists")
+    public void aValidPaymentExists(Map<String, String> params) {
+        testGatewayAccount.withExternalId(params.get("gateway_account_id")).insert(app.getTestContext().getJdbi());
+        testMandate.withGatewayAccountFixture(testGatewayAccount)
+                .withExternalId(MandateExternalId.valueOf(params.get("mandate_id")))
+                .withPayerFixture(PayerFixture.aPayerFixture())
+                .withMandateBankStatementReference(MandateBankStatementReference.valueOf(params.get("bank_mandate_reference")))
+                .withPaymentProviderId(GoCardlessMandateId.valueOf(params.get("unique_identifier")))
+                .withState(MandateState.PENDING)
+                .insert(app.getTestContext().getJdbi());
+        aPaymentFixture()
+                .withMandateFixture(testMandate)
+                .withExternalId("ch_123abc456def")
```

Comment:
i think you want this external id as part of the params. you'll need to put this charge_id in the publicapi pact file, see https://github.com/alphagov/pay-publicapi/blob/master/src/test/resources/pacts/publicapi-connector-get-payment-with-metadata.json#L16 as an example.